### DISCUSSION TITLE
Visual studio code integration

### DISCUSSION THREAD

**Discussion by thistlillo** [pieces-app/support#151] [Q&A]
I would like to use the Visual Studio Code integration of Pieces. I installed the Desktop app on Mac OS, then the extension in Visual Studio Code. Now, I would like to choose the Copilot:

<img width="288" alt="Screenshot 2024-04-09 at 11 17 46" src="https://github.com/pieces-app/support/assets/6717655/9975c3cb-9b05-42e2-a870-7a1d3376fa5f">

I try to register a custom API Key. When I click on it, I am asked to log in on the Pieces website. When I log in, I get a warning about a non-secure form, but accept to send the data anyway.

<img width="385" alt="Screenshot 2024-04-09 at 11 16 15" src="https://github.com/pieces-app/support/assets/6717655/e8ff7196-e7fa-4949-bd4c-0ed084e063f1">

Then I see this:

<img width="996" alt="Screenshot 2024-04-09 at 11 16 46" src="https://github.com/pieces-app/support/assets/6717655/c660321d-1705-4183-9cab-2fa0828cd5c5">

I give up on the OpenAI key and select PaLM 2 or Gemini. Nothing happens. Really nothing.

I could not find instructions: how am I supposed to change the Copilot?

versions:
mac os  14.4.1
pieces 2.11.0
visual studio code: 1.86.2
pieces for vs code: v1.14.1

**Comment by rosie-at-pieces** [comment]
hi @thistlillo, can you please [open up an issue ](https://github.com/pieces-app/support/issues/new/choose)for this?
